tests/cpp/CMakeLists.txt

```diff
@@ -13,6 +12,11 @@ if(NOT TARGET gtest)
   FetchContent_MakeAvailable(googletest)
 endif()
 
+FetchContent_Declare(safetensors.h
+    URL https://github.com/hsnyder/safetensors.h/archive/974a85d7dfd6e010558353226638bb26d6b9d756.tar.gz
+    URL_HASH SHA256=9aaf5961609601cf9aaa96582a207bce7c6e5fbf57ed2cc669bb7bde6a937d4b)
+FetchContent_MakeAvailable(safetensors.h)
```

Comment:
let's move `safetensors.h` downloading to `thirdparty/CMakeLists.txt` as well

Reply:
It was my initial intent, but lora_adapter can't find `safetensors.h`:
![image](https://github.com/user-attachments/assets/01b12bc4-9472-4b54-88f5-8b59c1cba685)